### ISSUE TITLE
pkg/v1/cache is unable to handle incomplete files

### ISSUE THREAD

**Issue by chrischdi** [google/go-containerregistry#717]
If using the cache and interrupting the download the following will happen:
* The in-flight layer stays as-is in the local cache directory
* On the next run, when [fsCache.Get] is getting called with the broken hash, `io.Copy` in [pkg/v1/hash.go](https://github.com/google/go-containerregistry/blob/master/pkg/v1/hash.go#L115) returns a `UnexpectedEOF` error (see below for full stacktrace)

This could get fixed by Deleting the broken file in [fsCache.Get] when `UnexpectedEOF` was returned and returning ErrNotFound, which will lead to downloading the file again.

I've pushed this change [on my fork here](https://github.com/chrischdi/go-containerregistry/commit/a264bb14a97d84343d53d887612fd4958a27ab53)

If this is valid I would be happy to open a PR for this :-)

Sidenote: I found this by using `container_pull` using the of [rules_docker](https://github.com/bazelbuild/rules_docker)

[fsCache.Get]: https://github.com/google/go-containerregistry/blob/master/pkg/v1/cache/fs.go#L103

Stacktrace to the UnexpectedEOF error
```
github.com/google/go-containerregistry/pkg/v1.SHA256 (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/hash.go:116)
github.com/google/go-containerregistry/pkg/v1/tarball.computeDigest (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/tarball/layer.go:144)
github.com/google/go-containerregistry/pkg/v1/tarball.LayerFromOpener (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/tarball/layer.go:113)
github.com/google/go-containerregistry/pkg/v1/tarball.LayerFromFile (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/tarball/layer.go:87)
github.com/google/go-containerregistry/pkg/v1/cache.(*fscache).Get (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/cache/fs.go:104)
github.com/google/go-containerregistry/pkg/v1/cache.(*image).Layers (/home/schloc/go/src/github.com/google/go-containerregistry/pkg/v1/cache/cache.go:64)
github.com/bazelbuild/rules_docker/container/go/pkg/compat.writeImageLayers (/home/schloc/go/src/github.com/bazelbuild/rules_docker/container/go/pkg/compat/write.go:120)
github.com/bazelbuild/rules_docker/container/go/pkg/compat.WriteImage (/home/schloc/go/src/github.com/bazelbuild/rules_docker/container/go/pkg/compat/write.go:150)
main.pull (/home/schloc/go/src/github.com/bazelbuild/rules_docker/container/go/cmd/puller/puller.go:103)
main.main (/home/schloc/go/src/github.com/bazelbuild/rules_docker/container/go/cmd/puller/puller.go:157)
runtime.main (/usr/local/go/src/runtime/proc.go:203)
runtime.goexit (/usr/local/go/src/runtime/asm_amd64.s:1373)
```

**Comment by ImJasonH** [comment]
This seems like a legitimate bug and your fix seems reasonable.

Could you add some kind of test that proves this works as expected, when you send the PR?

**Comment by chrischdi** [comment]
Sure :-) I've opened the PR and added a test prior to that :+1:

**Comment by or-shachar** [comment]
Hey :) Where do we stand with pushing the solution for this?  
Is there anything we can do in order to help this progress?

We've seen this issue on local dev envs and we had to create a script that manually cleans corrupted files from the cache folder.

**Comment by ImJasonH** [comment]
@or-shachar Thanks for pinging this, sorry this fell off my radar. The PR lgtm.

**Comment by ImJasonH** [comment]
Fixed by #720
